internal/ais/workflow.go

```diff
@@ -47,6 +44,23 @@ func (w *Workflow) Execute(ctx temporalsdk_workflow.Context, params *WorkflowPar
 	logger := temporalsdk_workflow.GetLogger(ctx)
 	logger.Debug("AIS workflow running!", "params", params)
 
+	defer func() {
+		logger.Debug("AIS workflow finished!", "result", r, "error", e)
```

Comment:
This could be misleading — if the workflow returns an error the returned result will *always* be nil. I've been tricked by this weird Temporal quirk a few times.

Reply:
Right, I'll make sure we also return a nil result if there is an error.